tests/Unit/ApparentHorizons/Test_ObserveCenters.cpp

```diff
@@ -0,0 +1,192 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <random>
+#include <string>
+
+#include "ApparentHorizons/ObserveCenters.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/Observer/Initialize.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/Tags.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Tags.hpp"
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/PrettyType.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+
+// This is needed to template ObserveCenters, but it doesn't have to be an
+// actual InterpolationTargetTag. This is only used for the name of the subfile
+// written
+struct AhA {};
+
+template <typename Metavariables>
+struct MockObserverWriter {
+  using component_being_mocked = observers::ObserverWriter<Metavariables>;
+  using replace_these_simple_actions = tmpl::list<>;
+  using with_these_simple_actions = tmpl::list<>;
+
+  using const_global_cache_tags =
+      tmpl::list<observers::Tags::ReductionFileName>;
+
+  using initialize_action_list =
+      tmpl::list<::Actions::SetupDataBox,
+                 observers::Actions::InitializeWriter<Metavariables>>;
+  using initialization_tags =
+      Parallel::get_initialization_tags<initialize_action_list>;
+
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockNodeGroupChare;
+  using array_index = int;
+
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        initialize_action_list>>;
+};
+
+struct TestMetavars {
+  using observed_reduction_data_tags = tmpl::list<>;
+
+  enum class Phase { Initialization, WriteData, Exit };
+
+  using component_list = tmpl::list<MockObserverWriter<TestMetavars>>;
+};
+
+using FoTPtr = std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>;
+
+SPECTRE_TEST_CASE("Unit.ApparentHorizons.ObserveCenters",
+                  "[Unit][ApparentHorizons]") {
+  using observer = MockObserverWriter<TestMetavars>;
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<double> center_dist{-10.0, 10.0};
+  const std::string filename = "ObserveCentersTest_Output";
+
+  // clean up just in case
+  if (file_system::check_if_file_exists(filename + ".h5")) {
```

Comment:
in other places we mock the writing action to avoid doing file IO during tests. Can you please switch to that approach? File IO in tests is a bit annoying, especially on supercomputers with slow filesystems